lib/rubocop/cop/rspec/let_before_examples.rb

```diff
@@ -64,6 +74,39 @@ def check_let_declarations(node)
             end
           end
         end
+
+        def find_first_example(node)
+          node.parent.children.find { |sibling| example_or_group?(sibling) }
+        end
+
+        def node_range(node)
+          range_between(node.loc.expression.begin_pos, last_node_loc(node))
+        end
+
+        def node_range_with_surrounding_space(node)
+          range = node_range(node)
+          range = range_with_surrounding_space(range, :left, false)
+          range = range_with_surrounding_space(range, :right, true)
+          range
+        end
+
+        def source(node)
+          node_range(node).source
+        end
+
+        def last_node_loc(node)
+          last_line = node.loc.end.line
+          heredoc_line(node) do |loc|
```

Comment:
Why are heredocs so hard? Isn’t there some shared code to be extracted that can help us deal with heredocs?

Reply:
Rubocop definitely needs some helpers to work with heredocs. I managed to simplify a bit